tools/d2g/d2gtest/testdata/testcases/container_engine_test.yml

```diff
@@ -0,0 +1,94 @@
+---
+testSuite:
+  name: Container engine tests
+  description: Test that the container engine should be set properly in the resulting generic worker task payload.
+  payloadTests:
+    - name: Default container engine test
+      description: >-
+        Tests that podman should be set as the default container engine in the resulting generic worker task payload.
+      dockerWorkerTaskPayload:
+        command:
+          - echo "Hello world"
+        image: ubuntu
+        maxRunTime: 3600
+      genericWorkerTaskPayload:
+        command:
+          - - bash
+            - '-cx'
+            - >-
+              podman run -t --rm --memory-swap -1 --pids-limit -1 --ulimit host
+              -e RUN_ID
+              -e TASKCLUSTER_INSTANCE_TYPE
+              -e TASKCLUSTER_ROOT_URL
+              -e TASKCLUSTER_WORKER_LOCATION
+              -e TASK_GROUP_ID
+              -e TASK_ID
+              ubuntu 'echo "Hello world"'
+        maxRunTime: 3600
+        onExitStatus:
+          retry:
+            - 125
+            - 128
+
+    - name: Default container engine test with privileged set
```

Comment:
in addition, one more test where privileged is set but containerEngine isn't set? or do we have that elsewhere? maybe not bad to have it here too, then we have covered all possibilities here